taichi/program/kernel.h

```diff
@@ -19,6 +19,7 @@ class TI_DLL_EXPORT Kernel : public Callable {
 
   bool is_accessor{false};
   bool is_evaluator{false};
+  bool has_real_func{false};
```

Comment:
Is `has_real_func` still needed? I can't find any usage of it in this PR.

Reply:
in [ir_node_extended_impl.cpp#30](https://github.com/taichi-dev/taichi/pull/4568/files#diff-ba3a420d15b12577ed1475815e2523183ee9afd8921445aadab7f4cc67a073b9R30)

Reply:
I mean is all those `has_real_func` and `set_has_real_func` functions and variables still needed?

Reply:
I use `Funcallstmt` instead of the check, let me remove it.